### ISSUE TITLE
New maintainer wanted

### ISSUE THREAD

**Issue by stachenov** [stachenov/quazip#185]
TL;DR: QuaZip is mostly abandoned until someone steps in as the new maintainer.

This project started a long time ago, circa 2005, when I needed something to work with ZIP archives in one of my Qt projects, and realized that no library is compatible with the Qt API. So I decided to make my own.

Since then, I've implemented all I've wanted, and the project became semi-popular. People actually started using it. Well, that's all good, that's what open source is for. I also kept using it, so I was still in the flow and could afford to spend some time maintaining it.

But then, something major happened. I changed my job (for the first time in 18 years!) and moved to another country at that. My life changed a lot, and among the changes are two things: I have much less free time now, and I barely ever deal with C/C++, I work mostly with Java and Kotlin now. As a consequence, even compiling and testing QuaZip before releasing it became a pain. I don't even have the necessary environment set up any more because I don't need it for anything else. And the thing with open source is, since I'm doing it for free, I'm not exactly motivated to sacrifice my valuable time and spend a lot of effort on something I no longer need myself.

So unless a new maintainer shows up, this project can be considered pretty much abandoned. Feel free to fork it and patch it for your own needs, of course, and if you'd like to become the new maintainer, well, just drop a comment here, I'll grant you the necessary rights.

**Comment by cen1** [comment]
I tried to contact some distro packagers and downstream users but haven't got much of a response to help out.

I am willing to try and maintain this for a bit longer, would hate to see it abandoned or forked into million pieces.

Hopefully someone else joins down the road.

**Comment by stachenov** [comment]
I've tried to add you to the collaborator list, check it out.

**Comment by cen1** [comment]
Looks ok, I can push. If anything  is missing I'll let you know.

**Comment by cen1** [comment]
@stachenov see if you can grant me permission to create and upload packages to this repo. I'd like to make images with different versions of Qt preinstalled to test the `QUAZIP_USE_QT_ZLIB=ON` variants which is not possible with GH Qt action.

**Comment by stachenov** [comment]
@cen1 I don't see this option anywhere. I never dealt with packages before, so I've no idea where to look. I've tried googling a bit, but without much success. If you could point me to a guide, it would help a lot.

**Comment by cen1** [comment]
According to docs I should already have that permission, I think I'm just naming the images wrong. Ignore for now.

**Comment by cen1** [comment]
@stachenov can you give me a brief instructions how to update the github pages docs?

**Comment by stachenov** [comment]
Check out the `gh-pages` branch into `doc/html`. Run `doxygen` to update these docs from sources. Then just commit and push the result.

**Comment by aikawayataro** [comment]
I'll try to check some issues in my spare time, and of course make a proper judgment over QtZlib.

**Comment by cen1** [comment]
@stachenov I am including OSSF scorecard scans to implement best security practices for the repo. Please protect the branches as reported here: https://github.com/stachenov/quazip/security/code-scanning/26

**Comment by stachenov** [comment]
OK, I've tried to protect `master` and `v0.9.x` from force-pushes and deletions. Is that enough?

The result is a bit weird, as the warning on the main repo page about the branch not being protected has disappeared, but in the repo settings it still warns me that there are no branch protection rules. At the same time, I can see the ruleset I've created under the rules section in the settings. It's either I did it correctly, but in a more complicated way than necessary (creating a ruleset instead of just enabling branch protection), or I did something wrong.

**Comment by cen1** [comment]
force push and deletion should be enough for now, it might be that the scanner does not know about the rules, I'll look into it

**Comment by stachenov** [comment]
The scanner seems to be detecting it all right, the score is now up to 3 from 0. It's the Branches section under repo settings that keeps telling me that I haven't protected any branches even though I have.

**Comment by cen1** [comment]
![Screenshot_20240401_175635](https://github.com/stachenov/quazip/assets/1158861/df0bc087-a968-438c-8210-9ff284f47ba8)
I tried to do it on my fork and this is how I see it

**Comment by geustache** [comment]
Hello stachenov 
Thank you for your contribution, which has been very useful in integrating zip archive management.
This is the first time I've contributed to an open source project and I don't really know how to contribute to this library.

What I can contribute is that I regularly compile and deploy my software (using quazip) in production under the Qt5 and Qt6.7.xx environments.
This under macOS / AppStore and Windows mingw

**Comment by cen1** [comment]
I'll use this issue as a general PSA for any in interested parties.

Since getting commit rights I didn't have a ton of time to work on quazip but it's better than nothing, some decent fixes landed and various pull requests have been quite helpful.

I hope to review #199 by the end of the year and then tag a "stable" release.

After that the priority task would be to update minizip source.

**Comment by stachenov** [comment]
Updating minizip source is easier said than done, because it's been modified quite a bit to adapt it to QuaZip's needs. And to implement certain features. I don't remember the details, but I've actually tried to look into it, and it doesn't look easy.

**Comment by cen1** [comment]
@stachenov please enable default code-ql on the repo: https://docs.github.com/en/code-security/code-scanning/enabling-code-scanning/configuring-default-setup-for-code-scanning

**Comment by stachenov** [comment]
Done. If I didn't mess up anything. It's setting up right now.

**Comment by mathewgrabau** [comment]
Is the project still looking for maintainers?

**Comment by cen1** [comment]
Additional maintainers are welcome. It would help to get some code review and second pair of eyes before merging so it's not a solo endeavor. There are also a few big and small issues that still need to be fixed/implemented.

Anyone who contributes (or did so in the past) a patch that solves something non-trivial is a good candidate in my books.

**Comment by mathewgrabau** [comment]
Thanks, @cen1 - I'll take a look and try to contribute to some of the issues. I would also be happy to contribute to code reviews.

**Comment by stachenov** [comment]
@cen1, I get a deja vu from this:

https://github.com/stachenov/quazip/issues/185#issuecomment-2699123573

As far as I can tell, you asked me to do exactly the same, and I did. Seems to be still enabled.

Or is that it's something else that you mean this time, and I fail to see it somehow?

**Comment by crueter** [comment]
@stachenov I am willing to step up as well since I use this in several of my projects. Most of my changes will probably be CI/build system upgrades, modernization, and maintenance.

**Comment by stachenov** [comment]
@crueter, OK, I've sent you an invite for colloboration.
